### PR TITLE
Nexus-22733:  Added update option handling for conan install command

### PR DESCRIPTION
**Pull request by igorudovika**
Added update option handling for conan install command

### CODE REVIEW

File: nexus-repository-conan/src/main/java/org/sonatype/repository/conan/internal/proxy/v1/ConanProxyFacet.java [hunk 1]
```diff
@@ -153,15 +156,16 @@ private Content putMetadata(final Context context,
     try (TempBlob tempBlob = storageFacet.createTempBlob(content.openInputStream(), HASH_ALGORITHMS)) {
       switch (assetKind) {
         case DOWNLOAD_URL:
+        case DIGEST:
           Content saveMetadata = doSaveMetadata(tempBlob, content, assetKind, coords);
```
Comment (mlukaretkyi) [inline nexus-repository-conan/src/main/java/org/sonatype/repository/conan/internal/proxy/v1/ConanProxyFacet.java]: Looks like if-else construction will look better
```
  if (assetKind == DOWNLOAD_URL || assetKind == DIGEST) {
        Content saveMetadata = doSaveMetadata(tempBlob, content, assetKind, coords);
        return new Content(
            new StringPayload(
                conanUrlIndexer.updateAbsoluteUrls(context, saveMetadata, getRepository()),
                ContentTypes.APPLICATION_JSON)
        );
  }
  return doSaveMetadata(tempBlob, content, assetKind, coords);
```

File: nexus-repository-conan/src/main/java/org/sonatype/repository/conan/internal/proxy/v1/ConanProxyControllerV1.java [hunk 2]
```diff
@@ -40,6 +41,7 @@
 
   public void attach(final Router.Builder builder) {
     createRoute(builder, ConanRoutes.downloadUrlsProxy(), DOWNLOAD_URL, proxyHandler);
+    createRoute(builder, ConanRoutes.digest(), DIGEST, proxyHandler);
     createRoute(builder, ConanRoutes.conanManifestProxy(), CONAN_MANIFEST, proxyHandler);
```
Comment (mlukaretkyi) [inline nexus-repository-conan/src/main/java/org/sonatype/repository/conan/internal/proxy/v1/ConanProxyControllerV1.java]: Please, rename to ConanRoutes.digest() -> ConanRoutes.digestProxy()

File: nexus-repository-conan/src/main/java/org/sonatype/repository/conan/internal/proxy/v1/ConanProxyFacet.java [hunk 3]
```diff
@@ -153,15 +156,16 @@ private Content putMetadata(final Context context,
     try (TempBlob tempBlob = storageFacet.createTempBlob(content.openInputStream(), HASH_ALGORITHMS)) {
       switch (assetKind) {
```
Comment (chernikovmaksym) [inline nexus-repository-conan/src/main/java/org/sonatype/repository/conan/internal/proxy/v1/ConanProxyFacet.java]: Consider using `if-else` here

File: nexus-repository-conan/src/main/java/org/sonatype/repository/conan/internal/proxy/v1/ConanProxyFacet.java (hunk 1 above)
Comment (igorudovika) [reply nexus-repository-conan/src/main/java/org/sonatype/repository/conan/internal/proxy/v1/ConanProxyFacet.java]: fixed 7e9eed00a7293a6b57a745c69ea24ddfc1a19f61

File: nexus-repository-conan/src/main/java/org/sonatype/repository/conan/internal/proxy/v1/ConanProxyControllerV1.java (hunk 2 above)
Comment (igorudovika) [reply nexus-repository-conan/src/main/java/org/sonatype/repository/conan/internal/proxy/v1/ConanProxyControllerV1.java]: Fixed 7e9eed00a7293a6b57a745c69ea24ddfc1a19f61

File: nexus-repository-conan/src/main/java/org/sonatype/repository/conan/internal/proxy/v1/ConanProxyFacet.java (hunk 3 above)
Comment (igorudovika) [reply nexus-repository-conan/src/main/java/org/sonatype/repository/conan/internal/proxy/v1/ConanProxyFacet.java]: Fixed 7e9eed00a7293a6b57a745c69ea24ddfc1a19f61

**Pull request merged**